contracts/protocol/libraries/logic/SupplyLogic.sol

```diff
@@ -128,14 +128,22 @@ library SupplyLogic {
 
     reserve.updateInterestRates(reserveCache, params.asset, 0, amountToWithdraw);
 
+    bool isCollateral;
+
+    if ((isCollateral = userConfig.isUsingAsCollateral(reserve.id))) {
+      if (amountToWithdraw == userBalance) {
+        userConfig.setUsingAsCollateral(reserve.id, false);
+        emit ReserveUsedAsCollateralDisabled(params.asset, msg.sender);
```

Comment:
here you should update cached `isCollateral` and set it into `false` as well

Reply:
what do you mean? There is no cached `userConfig`

Reply:
Checked the code one more time, my mistake, I originally thought that if we withdraw fully we should update `isCollateral` here as well. Nothing to fix